astropy/table/table.py

```diff
@@ -195,11 +203,28 @@ def as_array(self):
         if len(self.columns) == 0:
             return None
 
+        sys_byteorder = ('>', '<')[sys.byteorder == 'little']
```

Comment:
I didn't realize that you can use a bool index like this.  Nevertheless this seems like a job for the ternary conditional.